Standards/scs-XXXX-vN-mandatory-and-supported-IaaS-services.md

```diff
@@ -0,0 +1,77 @@
+---
+title: Mandatory and Supported IaaS Services
+type: Standard
+status: Draft
+track: IaaS
+---
+
+## Introduction
+
+To be SCS-compliant a Cloud Service Provider (CSP) has to fulfill all SCS standards.
+Some of those standards are broad and consider all APIs of all services on the IaaS-Layer like the consideration of a [role standard](https://github.com/SovereignCloudStack/issues/issues/396).
+There exist many services on that layer and for a first step they need to be limited to have a clear scope for the standards and the Cloud Service Providers following them.
+For this purpose, this standard will establish lists for mandatory services whose APIs have to be present in a SCS cloud as well as supported services, which APIs are considered by some standards and may even be tested for their integration but are optional in a sense that their omission will not violate SCS conformance.
+
+## Motivation
+
+There are many OpenStack APIs and their corresponding services that can be deployed on the IaaS level.
+These services have differences in the quality of their implementation and liveness and some of them may be easily omitted when creating an IaaS deployment.
+To fulfill all SCS-provided standards only a subset of these APIs are required.
+Some more but not all remaining OpenStack APIs are also supported additionally by the SCS project and may be part of its reference implementation.
+This results in different levels of support for specific services.
+This document will give readers insight about how the SCS classifies the OpenStack APIs accordingly.
+If a cloud provides all mandatory and any number of supported OpenStack APIs, it can be tested for SCS-compliance.
+Any unsupported APIs will not be tested.
+
+## Mandatory IaaS APIs
+
+The following IaaS APIs MUST be present in SCS-compliant IaaS deployments and could be implemented with the corresponding OpenStack services:
+
+| Mandatory API | corresponding OpenStack Service | description |
+|-----|-----|-----|
+| **block-storage** | Cinder | Block Storage service |
+| **compute** | Nova | Compute service |
+| **identity** | Keystone | Identity service |
+| **image** | Glance | Image service |
+| **load-balancer** | Octavia | Load-balancer service |
+| **network** | Neutron | Networking service |
+| **s3** or **object-store** | S3 API object storage | No formal standard exists, many implementations: Swift, RadosGW, minio... |
```

Comment:
I personally dislike "or" here. There are user loads supporting only one of both so having here an OR does not bring much benefit. S3 support in it's own is not standardized enough to have multiple compatibility issues, especially when it comes to the OpenStack Identity related stuff.

Reply:
We discussed this in the standardization SIG and following items were raised:
- S3 was always intended to be mandatory (also for KaaS layer)
- Swift should be at least "recommended"
Even though S3 and Swift both implement the same service (object-store) it can be seen as: HDMI output is a must and DisplayPort is recommended

Reply:
And it should be two separate rows

Reply:
I will change the standard accordingly. 
**BUT**, if S3 is also listed as "object-store" equal to Swift, we have no chance to distinguish between those two services in the tests. 
This is a little bit unfortunate, because we will never know for certain, whether the endpoints are from an S3 service or from Swift. So I can change the lines here, but we will not be able to test them!

An option may be to use gaia-x credentials. Which raises the question: Do we have a point, were we state which credentials we require? @garloff ?

Reply:
IMO, in this context, we need a standard for object storage too, similar like for block storage. CSPs need to know the requirements for object store and users need to know, which capabilities an object store has. See, #725

Reply:
I agree to @josephineSei: We can not distinguish between two object store APIs in tests. We should define just one as mandatory.  Furthermore, setting S3 as mandatory is  problematic. As far as I known, S3 API is not in control of OpenStack. There is no official documentation on how S3 for a specific OpenStack release looks like. The latest documentation is from [Mitaka](https://docs.openstack.org/mitaka/config-reference/object-storage/configure-s3.html).  There is no documentation for (current release)[https://docs.openstack.org/2024.1/config-reference/object-storage/configure-s3.html], e.g. Hence, we standardize a moving target IMO.

Reply:
There is still an open issue  with respect to version of a certain OpenStack service.  E.g. how does a consumer knows, which version of Nova is provided by a CSP? We will answer this question in #765